File: 프로그래머스/unrated/181862. 세 개의 구분자/세 개의 구분자.cpp
#include <string>
#include <vector>
#include <sstream>
#include <iostream>

using namespace std;

vector<string> solution(string myStr) {
    vector<string> answer;
    for(auto &str : myStr){
        if(str == 'a' || str == 'b' || str == 'c'){
            str = ' ';
        }
    }
    stringstream ss(myStr);
    string tkn = "";
    while(getline(ss, tkn, ' ')){
        if(tkn == "") continue;
        answer.push_back(tkn);
    }
    if(answer.size() == 0) answer.push_back("EMPTY");
    return answer;
}

File: 프로그래머스/unrated/181862. 세 개의 구분자/세 개의 구분자.java
class Solution {
    public String[] solution(String myStr) {
        String[] answer = {};
        char[] chrMyStr = myStr.toCharArray();
        for(int i=0;i<chrMyStr.length;i++){
            if(chrMyStr[i] == 'a' || chrMyStr[i] == 'b' || chrMyStr[i] == 'c')
                chrMyStr[i] = ' ';
        }
        myStr = String.valueOf(chrMyStr);
        myStr = myStr.replaceAll("\\s+", " ").trim();
        
        answer = myStr.split(" ");
        if(answer.length == 1 && answer[0].length() == 0){
            answer = new String[1];
            answer[0] = "EMPTY";
        }
        return answer;
    }
}